commands/general/stats.js

```diff
@@ -4,118 +4,165 @@ const moment = require('moment')
 
 module.exports.load = (client) => {
   client.commands['stats'] = {
-    run (message) {
-      let args = message.content.split(' ').splice(1)
-      let username
-      let discriminator
-      let mem
-      let userId
-      let av = './assets/default_avatar.jpg'
+    async run (message) {
+      let userInfo
+      try {
+        userInfo = selectTargetUser(message)
+      } catch (err) {
+        console.log(err.stack)
+        return client.errorMessage(message, `Unable to identify user: ${err}`)
+      }
+
+      try {
+        const [user, pos, tot, guild] = await Promise.all([
+          client.userRepository.load(userInfo.userId),
+          client.userRepository.getSessionPos(userInfo.userId),
+          client.userRepository.getSessionCount(),
+          client.guildRepository.load(message.guild.id)
+        ])
+        if (user !== null) {
+          const activeTime = getActiveTime(guild, userInfo.mem)
 
-      if (args.length >= 1) {
-        // fqName: "fully qualified name"
-        let fqName = args.join(' ').trim().split('#')
-        if (fqName.length === 2) {
-          username = fqName[0]
-          discriminator = fqName[1]
-          mem = client.guilds.get(message.guild.id).members.find(val => val.user.username === username && val.user.discriminator === discriminator)
-          if (mem != null) {
-            userId = mem.user.id
-            // checks if user has pfp because discord dosnt return default pfp url >:C
-            if (mem.user.avatarURL != null) {
-              av = mem.user.avatarURL
-            }
-          } else {
-            return client.errorMessage(message, `Unable to find user ${username}#${discriminator}.`)
-          }
+          userInfo.currentSession = user.current_session_playtime + activeTime
```

Comment:
I think I am slightly confused by the error conditions here. If I'm not mistaken, `getSessionPos` will return `undefined` if the user's time is zero, so we'll end up with a rank of NaN. (Right now we're detecting this condition in `fetchWeeklyLeaderboardPos` which simply returns a result `N / A`.)